### ISSUE TITLE
Does TinyTDS support SQL2016

### ISSUE THREAD

**Issue by markwintch** [rails-sqlserver/tiny_tds#311]
I am running Redmine 3.3.0 on Windows Server 2012 64 bit.  Rails version is 2.1.8 with Rails version 4.2.6

I cannot get this installation to talk to a newly built SQL2016 database.  So, my question is, do we have a tinytds gem compatible with SQL 2016 yet?

Thank you.

Mark

**Comment by metaskills** [comment]
> tinytds gem compatible with SQL 2016 yet?

Oh yeah. FreeTDS is great with this. We even test 2016 with [Appveyor](https://github.com/rails-sqlserver/tiny_tds/blob/master/appveyor.yml) here. You likely just have a mis configuration. Did you follow all the advice in the issue template when creating this one? https://github.com/rails-sqlserver/tiny_tds/blob/master/ISSUE_TEMPLATE.md

**Comment by markwintch** [comment]
Ken, I am sorry to have to ask such a basic question, but I am going a little crazy with this.  My Redmine installation currently has tiny_tds version 0.6.2-x64-mingw32  I am trying to update to any version that will support SQL2016 as when I execute the rake db:migrate command when connected to the SQL 2016 database I get the specific error message that SQL 2016 is not supported.  Here is the rookie part that I apologize for.  I CANNOT get version 1.0.5 to install.  I downloaded the gem and placed it in my gems directory and tried the gem install tiny_tds -n {directorypath}  this failed, I tried gem install tiny_tds -v1.0.5, gem install tiny_tds -l I uninstalled the gem and then also uninstalled the file.  So here is my question?  What version of tiny_tds do I need to install and how do I do that?

Redmine version  2.6.5
Rails Version 3.2.21
Ruby Version 2.1.6-p336 [x64-mingw32]
Database adaptor: SQLServer

Repeating question in the event you skimmed or skipped my description above:  What tiny_tds version do I need and how do I install it into Redmine to use a SQL 2016 database?

Thank you again very much!

Mark

**Comment by YvesR** [comment]
Hello @markwintch,
your problem is not tinyTDS here, it is probably activerecord-sqlserver-adapter.
There you get "not supported" message for SQL Server 2016 if it is not in the version list.
Try checkout latest gem here or modify the supported version list in the gem yourself.

**Comment by metaskills** [comment]
Glad to help @markwintch and yes @YvesR is right. Back in the day we thought it was a good idea to check the DB version and we found it caused more issues than it helped with because it is almost always cool to use an older adapter/tinytds with a NEWER version of SQL Server. We removed these checks in the Rails 4.2 and upward. If you are curious, here is the error you are hitting in 3.2.x

https://github.com/rails-sqlserver/activerecord-sqlserver-adapter/blob/3-2-stable/lib/active_record/connection_adapters/sqlserver_adapter.rb#L184-L224

To make yours and others life more sane, I've pushed adapter gem updates that add 2016 to the list. But please, do try to use the latest 4.x. Either way, any latest 3.2.x, 4.0.x, 4.1.x, or 4.2.x adapter should work just fine for you now.

**Comment by metaskills** [comment]
Also, just realized that since I made the changes and pushed new gem, that link above will now show 2016 in the supported list. Just FYI.

**Comment by markwintch** [comment]
I did start attempting to install Tiny_tds 1.0.5 and got the dependencies
errors.  I think I have all of the gems I need except for one active?
something.  I requires Ruby 2.2, So, I will dig into this and try to
resolve all of the dependencies without breaking my dev environment.  That
is the reason we have a dev environment though.  Here is the list I have
compiled thus far.  If anyone wants I will submit the "guide" I put
together for this upgrade to SQL2016 for us.

activemodel-5.0.0.1
activerecord-5.0.0.1
active-sqlserver-adapter-4.2.15
activesupport-5.0.0.1
concurrent-ruby-1.0.2
mini_portile2-2.1.0
minitest-5.9.1
thread-safe-0.3.5
tiny_tds-1.0.5-x64-mingw32
tzinfo-1.2.2

ruby2.2.2

Thank you very much for your rapid answers each time!

Mark

*Mark Wintch*

Systems Analyst I



*Phone:* 913-309-4503

*Fax:* 1-866-470-0066

*Email:* mwintch@grantham.edu


 [image: Inline image 1]

       Online.* But not alone*.*TM*



Know someone who should hear about Grantham?  Refer a Friend
<http://grantham.edu/refer-a-friend>!

**Comment by metaskills** [comment]
> activemodel-5.0.0.1

This looks like you went to Rails v5. The SQL Server adapter is not Rails 5 ready yet. Please use Rails 4.2.x

**Comment by markwintch** [comment]
OK, I will backtrack.  As I started down this trail interdependencies were
requiring some of the versions I selected.  Does anyone happen to have a
list of exactly what minimum gems I will need to work with SQL2016?

Thank you!

*Mark Wintch*

Systems Analyst I



*Phone:* 913-309-4503

*Fax:* 1-866-470-0066

*Email:* mwintch@grantham.edu


 [image: Inline image 1]

       Online.* But not alone*.*TM*



Know someone who should hear about Grantham?  Refer a Friend
<http://grantham.edu/refer-a-friend>!
